clang/test/Driver/sycl-specific-args-diagnostics.cpp

```diff
@@ -0,0 +1,167 @@
+
+// Warning should be emitted when using -fsycl-libspirv-path without -fsycl
+// RUN:   %clang -### -fsycl-libspirv-path=libspirv.bc  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-LIBSPRIV %s
+// WARNING-LIBSPRIV: warning: argument unused during compilation: '-fsycl-libspirv-path=libspirv.bc' [-Wunused-command-line-argument]
+// RUN:   %clang_cl -### -fsycl-libspirv-path=libspirv.bc  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-LIBSPRIV-CL %s
+// WARNING-LIBSPRIV-CL: warning: argument unused during compilation: '-fsycl-libspirv-path=libspirv.bc' [-Wunused-command-line-argument]
+
+// Warning should be emitted when using -fsycl-max-parallel-link-jobs without -fyscl
+// RUN:   %clang -### -fsycl-max-parallel-link-jobs=4  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-PARALLEL %s
+// WARNING-PARALLEL: warning: argument unused during compilation: '-fsycl-max-parallel-link-jobs=4' [-Wunused-command-line-argument]
+// RUN:   %clang_cl -### -fsycl-max-parallel-link-jobs=4  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-PARALLEL-CL %s
+// WARNING-PARALLEL-CL: warning: argument unused during compilation: '-fsycl-max-parallel-link-jobs=4' [-Wunused-command-line-argument]
+
+// Warning should be emitted when using -fsycl-optimize-non-user-code without -fsycl
+// RUN:   %clang -### -fsycl-optimize-non-user-code  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-OPTIMIZE %s
+// WARNING-OPTIMIZE: warning: argument unused during compilation: '-fsycl-optimize-non-user-code' [-Wunused-command-line-argument]
+// RUN:   %clang_cl -### -fsycl-optimize-non-user-code  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-OPTIMIZE-CL %s
+// WARNING-OPTIMIZE-CL: warning: argument unused during compilation: '-fsycl-optimize-non-user-code' [-Wunused-command-line-argument]
+
+// Warning should be emitted when using -fsycl-unnamed-lambda without -fsycl
+// RUN:   %clang -### -fsycl-unnamed-lambda  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNNAMED-LAMBDA %s
+// WARNING-UNNAMED-LAMBDA: warning: argument unused during compilation: '-fsycl-unnamed-lambda' [-Wunused-command-line-argument]
+// RUN:   %clang_cl -### -fsycl-unnamed-lambda  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNNAMED-LAMBDA-CL %s
+// WARNING-UNNAMED-LAMBDA-CL: warning: argument unused during compilation: '-fsycl-unnamed-lambda' [-Wunused-command-line-argument]
+
+// Warning should be emitted when using -fsycl-use-bitcode without -fsycl
+// RUN:   %clang -### -fsycl-use-bitcode  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-USE-BITCODE %s
+// WARNING-USE-BITCODE: warning: argument unused during compilation: '-fsycl-use-bitcode' [-Wunused-command-line-argument]
+// RUN:   %clang_cl -### -fsycl-use-bitcode  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-USE-BITCODE-CL %s
+// WARNING-USE-BITCODE-CL: warning: argument unused during compilation: '-fsycl-use-bitcode' [-Wunused-command-line-argument]
```

Comment:
For any diagnostic checks performed, instead of using multiple string checks based on the specific option, you can do something like:
// RUN: %clang_cl -### -fsycl-use-bitcode %s 2>&1 \
// RUN:  | FileCheck -check-prefix=WARNING_CHECK -DOPT=-fsycl-use-bitcode %s
// WARNING_CHECK: warning: argument unused during compilation: '[[OPT]]' [-Wunused-command-line-argument]